### DISCUSSION TITLE
Why is Vulkan not the default render mode for Android?

### DISCUSSION THREAD

**Discussion by JakeIn2l** [AvaloniaUI/Avalonia#16717] [Q&A] [answered]
Hello,

I found that in my tests, setting the rendering mode to Vulkan lends to a much smoother app than the default rendering mode. Looking at the code, the default is EGL, with a fallback to software rendering. I tested with both rendering modes with AOT compilation.

I was wondering why this was the default behavior, and if I'm missing something?

**Comment by maxkatz6** [comment, marked as answer]
Because it was only added recently and isn't yet well tested.
It might be a default soon.

**Reply by dbriard** [reply]
@maxkatz6 I have an aditionnal question if you don't mind.

I often use ICustomDrawOperation in my code to do complex rendering that I cannot do directly using Avalonia api (blur, image effects, shaders, etc...).
Does the rendering engine affect what I can do using SkiaSharp in ICustomDrawOperation?
I mean if tomorrow Vulkan is the rendering mode, are custom SkiaSharp calls will continue to work?
Thanks.

**Reply by maxkatz6** [reply]
If you use ICustomDrawOperation to render custom content, it's all up to Skia at this point.

Most likely your app will be fine, if you migrate from OpenGL to Vulkan on android, for example, as Skia maps its own shaders on both of these rendering engines. Switching back to software renderer might break some functionality in theory, if Skia won't be able to emulate them.

**Reply by dbriard** [reply]
Thx, that's what I was hoping for :)

**Comment by Coloryr** [comment]
`Vulkan` can be `default` with drm or fbdev?
